tests/Feature/StoreTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use App\Models\Store;
+
+it('user can add a new store', function () {
+
+    userLogin();
+
+    $this->post(route('store.store'), [
+        'name' => '2020',
+    ])
+        ->assertStatus(302)
+        ->assertRedirect(route('store.index'));
+
+    $this->assertDatabaseHas('stores', [
+        'name' => '2020',
+    ]);
+});
+
+it('user can update a store', function () {
+
+    userLogin();
+
+    $store = Store::factory()->create(['name' => '2022']);
+
+    $this->post(route('store.update', $store->id), [
+        'id' => $store->id,
+        'name' => 'update 2022'
+    ])
+        ->assertRedirect(route('store.index'));
+
+    $this->assertDatabaseHas('stores', [
+        'id' => $store->id,
+        'name' => 'update 2022',
+    ]);
+});
+
+it('user can delete a store', function () {
```

Comment:
add test `user can not visit the stores page without login`

Reply:
return view('admin.pages.store.form', ['store' => $store]); sorry sir store ave che